.github/actions/k-bench/evaluate/graph.py

```diff
@@ -0,0 +1,193 @@
+    """Draws a bar chart with multiple bars per data point.
+
+    Args:
+        data (dict[str, list]): Benchmark data dictionary: subject -> lists of value points
+        headers (list): List of headers (x-axis).
+        title (str, optional): The title for the chart. Defaults to "".
+        suffix (str, optional): The suffix for values e.g. "MiB/s". Defaults to "".
+        val_label (bool, optional): Put a label of the value over the bar chart. Defaults to True.
+        y_log (bool, optional): Set the y-axis to a logarithmic scale. Defaults to False.
+    Returns:
+        fig (matplotlib.pyplot.figure): The pyplot figure
+    """
+    fig, ax = plt.subplots(figsize=(10, 5))
+    fig.patch.set_facecolor('white')
+
+    # Number of bars per group
+    n_bars = len(data)
+
+    # The width of a single bar
+    bar_width = 0.8 / n_bars
+
+    # List containing handles for the drawn bars, used for the legend
+    bars = []
+
+    # Iterate over all data
+    for i, values in enumerate(data.values()):
+        # The offset in x direction of that bar
+        x_offset = (i - n_bars / 2) * bar_width + bar_width / 2
+
+        # Draw a bar for every value of that type
+        for x, y in enumerate(values):
+            bar = ax.bar(x + x_offset, y, width=bar_width * 0.9,
+                         color=BAR_COLORS[i % len(BAR_COLORS)], edgecolor='black')
+            if val_label:
+                ax.bar_label(bar, fmt='%g {suffix}'.format(suffix=suffix))
+        # Add a handle to the last drawn bar, which we'll need for the legend
+        bars.append(bar[0])
+    # Draw legend
+    ax.legend(bars, LEGEND_NAMES)
+    if y_log:
+        ax.set_yscale('log')
+    ax.set_xticks(np.arange(len(headers)))
+    ax.set_xticklabels(headers)
+
+    plt.setp(ax.get_xticklabels(), fontsize=LABEL_FONTSIZE,
+             rotation=LABEL_ROTATE_BY)
+    plt.title('{title}({suffix})'.format(title=title, suffix=suffix))
+    plt.tight_layout()
+    return fig
+
+
+def main():
+    """Read the files and create diagrams."""
+    out_dir = configure()
+    combined_results = defaultdict(dict)
+
+    for test in SUBJECTS:
+        # Read the previous results
+        read_path = os.path.join(
+            out_dir, '{subject}.json'.format(subject=test))
+        try:
+            with open(read_path, 'r') as res_file:
+                combined_results[test].update(json.load(res_file))
+        except OSError as e:
+            raise ValueError(
+                'Failed reading {subject} benchmark records: {e}'.format(subject=test, e=e))
+
+    # Combine the evaluation of the Kubernetes API benchmarks
+    for i, api in enumerate([pod_key2header, svc_key2header, depl_key2header]):
+        api_data = {}
+        for s in SUBJECTS:
+            points = combined_results[s]['kbench']
+            subject_data = [points[h] for h in api]
+            api_data[s] = subject_data
+        hdrs = list(api.values())
+        bar_chart(data=api_data, headers=hdrs,
```

Comment:
Nit: A space would be good so that:
"API Latency(ms)" -> "API Latency (ms)"

(same for the other graphs)